File: WaysOfLooking/W04/Java-Concrete-Fish/RockPaperScissors.java
public class RockPaperScissors {
    static int CalculateRoundScore(char opponentChoice, char playerChoice) {
        int playerScore = 0;
        int outcomeScore = 0;
        switch (playerChoice) {
            case 'X':
                playerScore = 1;
                break;
            case 'Y':
                playerScore = 2;
                break;
            case 'Z':
                playerScore = 3;
                break;
        }
        if (IsWin(opponentChoice, playerChoice)) {
            outcomeScore = 6;
        } else if (IsLoss(opponentChoice, playerChoice)) {
            outcomeScore = 0;
        } else {
            outcomeScore = 3;
        }
        return playerScore + outcomeScore;
    }

    static boolean IsWin(char opponentChoice, char playerChoice) {
        switch (opponentChoice) {
            case 'A':
                return playerChoice == 'Y';
            case 'B':
                return playerChoice == 'Z';
            case 'C':
                return playerChoice == 'X';
            default:
                return false;
        }
    }

    static boolean IsLoss(char opponentChoice, char playerChoice) {
        switch (opponentChoice) {
            case 'A':
                return playerChoice == 'Z';
            case 'B':
                return playerChoice == 'X';
            case 'C':
                return playerChoice == 'Y';
            default:
                return false;
        }
    }

    static int CalculateGameScore(String strategyGuide) {
        int totalScore = 0;
        for (int i = 0; i < strategyGuide.length(); i += 2) {
            char opponentChoice = strategyGuide.charAt(i);
            char playerChoice = strategyGuide.charAt(i + 1);
            int roundScore = CalculateRoundScore(opponentChoice, playerChoice);
            totalScore += roundScore;
            System.out.println("Round Score: " + roundScore);
        }
        return totalScore;
    }

    public static void main(String[] args) {
        String strategyGuide = "AYBXCZ";
        int totalScore = CalculateGameScore(strategyGuide);
        System.out.println("Total Score: " + totalScore);
    }
}

File: WaysOfLooking/W09/Fishing-Pole/RockPaperScissors.go
package main

import (
	"fmt"
)

const (
	playerRockCode = "X"
	opponentRockCode = "A"
	playerPaperCode = "Y"
	opponentPaperCode = "B"
	playerScissorsCode = "Z"
	opponentScissorsCode = "C")


 func calculateGameScore(strategyGuide string) int {
	var totalScore int
	for i := 0; i < len(strategyGuide); i += 2 {
		opponentChoice := string(strategyGuide[i])
		playerChoice := string(strategyGuide[i+1])
		outcomeScore := calculateRoundScore(opponentChoice, playerChoice)

		choiceScore := calculateChoiceScore(playerChoice)
		roundScore := choiceScore + outcomeScore
		fmt.Printf("Round Score: %v\r\n", roundScore)
		totalScore += roundScore
	}
	return totalScore
}
    
func calculateRoundScore(opponentChoice string, playerChoice string) int {
	if playerWins(opponentChoice, playerChoice) {
		return 6
	} else if playerLosses(opponentChoice, playerChoice) {
		return 0
	} else {
		return 3
	}
}

func calculateChoiceScore(playerChoice string) int {
		if playerChoice == playerRockCode {
		return 1
	} else 	if playerChoice == playerPaperCode {
		return 2
	} else 	if playerChoice == playerScissorsCode {
		return 3
	} else  {
		return 0
	}
}

func playerWins(opponentChoice string, playerChoice string) bool {
  if (opponentChoice == "A" && playerChoice == "Y") ||
(opponentChoice == "B" && playerChoice == "Z") ||
(opponentChoice == "C" && playerChoice == "X") {
   return true
  }
  return false
}

func playerLosses(opponentChoice string, playerChoice string) bool {
  if (opponentChoice == "B" && playerChoice == "X") ||
(opponentChoice == "C" && playerChoice == "Y") ||
(opponentChoice == "A" && playerChoice == "Z") {
    return true
  }
  return false
}
func main() {
	strategyGuide := "AYBXCZ"
	totalScore := calculateGameScore(strategyGuide)
	fmt.Printf("Total Score: %v", totalScore)
}
